### PR TITLE
Use y_train_.mean() instead of private attribute

### PR DESCRIPTION
**Pull request by jnothman**
It was a dare.

### CODE REVIEW

**Comment by betatim** [comment]
Should we review this?

Maybe we should add some comments to the line of code because to me it is not clear why it causes so many test failures :-/

**Comment by betatim** [comment]
I think the dare resulted from https://github.com/scikit-optimize/scikit-optimize/issues/462#issuecomment-320499879 (I am a bit slow today :)

**Comment by glouppe** [comment]
It is more complicated than that, as we should also deal with the `normalize_y` logic I suppose, as in https://github.com/scikit-learn/scikit-learn/blob/ec0dbf0399c4187a1d17e66de17265c67588373f/sklearn/gaussian_process/gpr.py#L183

**Comment by jnothman** [comment]
Right, thanks. I'd not checked the tests.

**Comment by jnothman** [comment]
You mean you want it to store zeros if `not normalize_y`

**Comment by jnothman** [comment]
This change isn't passing tests locally. I might leave it to you folks to fix.
